flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/SpillableSubpartition.java

```diff
@@ -130,16 +131,12 @@ public void release() throws IOException {
 				spillWriter.closeAndDelete();
 			}
 
-			// Get the view...
-			view = readView;
-			readView = null;
-
 			isReleased = true;
 		}
 
 		// Release the view outside of the synchronized block
-		if (view != null) {
-			view.notifySubpartitionConsumed();
+		if (!readView.isEmpty()) {
```

Comment:
Because this is discarding the whole partition, we should do this for all elements of the queue.